modules/linear_boltzmann_solvers/discrete_ordinates_solver/sweep/communicators/aah_async_comm.cc

```diff
@@ -70,192 +62,157 @@ AAH_ASynchronousCommunicator::Reset()
+  delayed_preloc_msg_received_.resize(num_delayed_dependencies);
 
-  delayed_prelocI_message_count_.resize(num_delayed_dependencies, 0);
-  delayed_prelocI_message_size_.resize(num_delayed_dependencies);
-  delayed_prelocI_message_blockpos_.resize(num_delayed_dependencies);
-  delayed_prelocI_message_received_.clear();
-
-  for (int prelocI = 0; prelocI < num_delayed_dependencies; ++prelocI)
+  for (int i = 0; i < num_delayed_dependencies; ++i)
   {
-    size_t num_unknowns = fluds.GetDelayedPrelocIFaceDOFCount(prelocI) * num_groups_ * num_angles_;
-    size_t message_count = num_angles_;
-    if (num_unknowns * 8 > max_mpi_message_size_)
-      message_count = ((num_unknowns * 8) + (max_mpi_message_size_ - 1)) / max_mpi_message_size_;
-    size_t message_size = (num_unknowns + (message_count - 1)) / message_count;
-
-    delayed_prelocI_message_count_[prelocI] = message_count;
-    delayed_prelocI_message_size_[prelocI].reserve(message_count);
-    delayed_prelocI_message_blockpos_[prelocI].reserve(message_count);
+    size_t num_unknowns = fluds.GetDelayedPrelocIFaceDOFCount(i) * num_groups_ * num_angles_;
+    auto [message_count, message_size] = message_count_and_size(num_unknowns);
+    const int locJ = spds.GetDelayedLocationDependencies()[i];
+    const auto source = comm_set_.MapIonJ(locJ, opensn::mpi_comm.rank());
 
     size_t pre_block_pos = 0;
+    delayed_preloc_msg_data_[i].reserve(message_count);
     for (int m = 0; m < message_count - 1; ++m)
     {
-      delayed_prelocI_message_size_[prelocI].push_back(message_size);
-      delayed_prelocI_message_blockpos_[prelocI].push_back(pre_block_pos);
+      delayed_preloc_msg_data_[i].emplace_back(
+        std::make_tuple(source, message_size, pre_block_pos));
       num_unknowns -= message_size;
       pre_block_pos += message_size;
     }
     if (num_unknowns > 0)
-    {
-      delayed_prelocI_message_blockpos_[prelocI].push_back(pre_block_pos);
-      delayed_prelocI_message_size_[prelocI].push_back(num_unknowns);
-    }
+      delayed_preloc_msg_data_[i].emplace_back(
+        std::make_tuple(source, num_unknowns, pre_block_pos));
+    else
+      --message_count;
 
-    delayed_prelocI_message_received_.emplace_back(message_count, false);
+    delayed_preloc_msg_received_[i].resize(message_count, false);
+    max_num_messages_ = std::max(message_count, max_num_messages_);
   }
 
   // Successor locations
-  size_t num_successors = spds.GetLocationSuccessors().size();
-
-  deplocI_message_count_.resize(num_successors, 0);
-  deplocI_message_size_.resize(num_successors);
-  deplocI_message_blockpos_.resize(num_successors);
-  deplocI_message_request_.clear();
+  const auto& location_successors = spds.GetLocationSuccessors();
+  const size_t num_successors = location_successors.size();
+  size_t total_deploc_messages = 0;
+  deploc_msg_data_.resize(num_successors);
 
-  for (int deplocI = 0; deplocI < num_successors; ++deplocI)
+  for (int i = 0; i < num_successors; ++i)
```

Comment:
The same here: `size_t` instead of `int`

Reply:
Fixed.